modules/insticatorBidAdapter.js

```diff
@@ -0,0 +1,272 @@
+    },
+    device: buildDevice(),
+    regs: buildRegs(bidderRequest),
+    user: buildUser(),
+    imp: validBidRequests.map((bidRequest) => buildImpression(bidRequest)),
+  };
+
+  const params = config.getConfig('insticator.params');
+
+  if (params) {
+    req.ext = {
+      insticator: params,
+    };
+  }
+
+  return req;
+}
+
+function buildBid(bid, bidderRequest) {
+  const originalBid = bidderRequest.bids.find((b) => b.bidId === bid.impid);
+
+  return {
+    requestId: bid.impid,
+    creativeId: bid.crid,
+    cpm: bid.price,
+    currency: 'USD',
+    netRevenue: true,
+    ttl: bid.exp || config.getConfig('insticator.bidTTL') || BID_TTL,
+    width: bid.w,
+    height: bid.h,
+    mediaType: 'banner',
+    ad: bid.adm,
+    adUnitCode: originalBid.adUnitCode,
+  };
+}
+
+function buildBidSet(seatbid, bidderRequest) {
+  return seatbid.bid.map((bid) => buildBid(bid, bidderRequest));
+}
+
+function validateSize(size) {
+  return (
+    size instanceof Array &&
+    size.length === 2 &&
+    typeof size[0] === 'number' &&
+    typeof size[1] === 'number'
+  );
+}
+
+function validateSizes(sizes) {
+  return (
+    sizes instanceof Array &&
+    sizes.length > 0 &&
+    sizes.map(validateSize).reduce((a, b) => a && b, true)
+  );
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  isBidRequestValid: function (bid) {
+    if (!bid.params.adUnitId) {
+      logError('insticator: missing adUnitId bid parameter');
+      return false;
+    }
+
+    if (!(BANNER in bid.mediaTypes)) {
+      logError('insticator: expected banner in mediaTypes');
+      return false;
+    }
+
+    if (
+      !validateSizes(bid.sizes) &&
+      !validateSizes(bid.mediaTypes.banner.sizes)
+    ) {
+      logError('insticator: banner sizes not specified or invalid');
+      return false;
+    }
+
+    return true;
+  },
+
+  buildRequests: function (validBidRequests, bidderRequest) {
+    const requests = [];
+
+    if (validBidRequests.length > 0) {
+      requests.push({
+        method: 'POST',
+        url: config.getConfig('insticator.endpointUrl') || ENDPOINT,
+        options: {
+          contentType: 'application/json',
+          withCredentials: true,
+        },
+        data: JSON.stringify(buildRequest(validBidRequests, bidderRequest)),
+        bidderRequest,
+      });
+    }
+
+    return requests;
+  },
+
+  interpretResponse: function (serverResponse, request) {
+    const bidderRequest = request.bidderRequest;
+    const body = serverResponse.body;
+
+    if (!body || body.id !== bidderRequest.bidderRequestId) {
+      logError('insticator: response id does not match bidderRequestId');
+      return [];
+    }
+
+    if (!body.seatbid) {
+      return [];
+    }
+
+    const bidsets = body.seatbid.map((seatbid) =>
+      buildBidSet(seatbid, bidderRequest)
+    );
+
+    return bidsets.reduce((a, b) => a.concat(b), []);
+  },
+
+  getUserSyncs: function (options, responses) {
```

Comment:
Do you wanna check here for Gdpr and Usp consents?

Reply:
We check the GDPR and USP consent on the backend side and pass them via ext.sync.